code/modules/mob/living/carbon/human/species.dm

```diff
@@ -255,50 +256,61 @@ GLOBAL_LIST_EMPTY(features_by_species)
 
 	if(heart && (!should_have_heart || replace_current))
 		heart.Remove(C,1)
+		required_organs -= /obj/item/organ/heart
 		QDEL_NULL(heart)
 	if(should_have_heart && !heart)
 		heart = new mutant_heart()
 		heart.Insert(C)
+		required_organs += /obj/item/organ/heart
```

Comment:
change all of them to `|=`, not `+=`
`|=` means "check if it exists in the list. If it exists, skips it. If not, add it.", so it's good when `required_organs` list can have unintended multiple same organ